walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,59 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Arrays;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<TimeRange> times = new ArrayList<TimeRange>();
+    int beginning = TimeRange.START_OF_DAY;
+    int end = TimeRange.END_OF_DAY;
+    TimeRange middle_current;
+    TimeRange middle_past = TimeRange.fromStartDuration(TimeRange.END_OF_DAY, 0);
+    //check for primary edge cases of duration longer than one day
+    if (request.getDuration() > (24*60)){
+        return Arrays.asList();
+    //check for primary edge cases of empty attendee list
+    } else if (request.getAttendees().isEmpty()) {
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    //loop through events updating middle and beginning times
+    //think of it as a line .----------. each time an event is added 
+    //we add a new dot and update the beginnign and middle pointers
+    // .-----.-----.
+    } else {
+        for (Event event : events) {
+            middle_current = event.getWhen();
+            //check if the two events are overlapped
+            if (beginning > middle_current.start()) {
+                //if they are overlapped then make the next avaliable slot
+                //the time of whichever ends later
+                if (middle_current.end() > middle_past.end()) {
+                    beginning = middle_current.end();
+                } else {
+                    beginning = middle_past.end();
+                }
+            //if they aren't overlapped then just update middle and beginning point
+            } else {
+                //make sure the request and the meetings Attendees are the same
+                if (request.getAttendees().containsAll(event.getAttendees()) == false) {
```

Comment:
If A is requested, and B is attending the current event, you will just return the whole day here.

Also I think `if (bool)` works -- no need for `if (bool == true)`

Reply:
Thank you for all the comments Hidy! I'll fix all of that right now.